### PR TITLE
Laughter demons now eject spontaneous revivals

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
tweak: If you are luckily enough to revive while still inside the
stomach of a laughter demon, you will be immediately ejected. Whether
you will be able to get away before the demon then kills and eats you
again, is up to you.
/:cl:

Very occasionally, a laughter demon will eat a changeling, or someone
with very strong nanites, and then are rendered helpless as someone
punches their way out.

Instead, now they will be confusingly ejected from the laughter demon's
internal hammerspace pocket, where they can than then attempt to make
their escape.

### CODE REVIEW

**Comment by D4C-420** [comment]
his grace should have something similar

**Comment by Tlaltecuhtli** [comment]
give it to other vore animals like dragon

**Comment by cacogen** [comment]
Wait I thought vore was removed

**Comment by ilikesaying** [comment]
vore can never be removed

**Comment by SkeletalElite** [comment]
> Wait I thought vore was removed

When dragons hit a corpse they vore it and heal. They drop the corpses on death. They probably have the same problem with revivals that (s)laughter demons have.

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 1]
```diff
@@ -259,21 +259,32 @@
...
-		// Keep their corpse so rescue is possible
-		consumed_mobs += victim
-	else
-		// Be safe and just eject the corpse
-		victim.forceMove(get_turf(victim))
-		victim.exit_blood_effect()
-		victim.visible_message("<span class='warning'>[victim] falls out of the air, covered in blood, looking highly confused. And dead.</span>")
+	// Keep their corpse so rescue is possible
+	consumed_mobs += victim
+	RegisterSignal(victim, COMSIG_MOB_STATCHANGE, .proc/on_victim_statchange)
```
Comment (Rohesie) [inline code/modules/antagonists/slaughter/slaughter.dm:274]:
```suggestion
	if(isliving(victim))
		RegisterSignal(victim, COMSIG_MOB_STATCHANGE, .proc/on_victim_statchange)
```

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 2]
```diff
@@ -259,21 +259,32 @@
...
+
+/mob/living/simple_animal/hostile/imp/slaughter/laughter/proc/on_victim_statchange(mob/victim, new_stat)
+	if(!isliving(victim))
+		return
+
```
Comment (Rohesie) [inline code/modules/antagonists/slaughter/slaughter.dm:279]: The victim is now guaranteed to be living.
Document this proc and use the signal handler macro, please.

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 3]
```diff
@@ -259,21 +259,32 @@
...
+	var/mob/living/living_victim = victim
+	if(new_stat != DEAD)
```
Comment (Rohesie) [inline code/modules/antagonists/slaughter/slaughter.dm:281]:
```suggestion
	if(victim.stat != DEAD || new_stat == DEAD)
		return
```
* Early return
* You want to expel them if they transition from dead to living, but not if they fall to crit or w/e (is that even possible?)

**Review by Cyberboss — CHANGES_REQUESTED** [review]
What @Rohesie said + changelog entries don't mix with newlines IIRC

**Pull request merged**